pipert/core/message_handlers.py

```diff
@@ -48,14 +74,51 @@ def __init__(self, url, maxlen=100):
         self.conn = None
         self.url = url
         self.maxlen = maxlen
+        self.last_msg_id = None
         self.connect()
 
+    def read_next_msg(self, in_key):
+        return self._read_from_redis_using_method(
+            in_key=in_key,
+            reading_method=self.conn.xrange,
+            name=in_key,
+            count=1,
+            min=self._add_offset_to_stream_id(self.last_msg_id, 1)
+        )
+
+    def read_most_recent_msg(self, in_key):
+        return self._read_from_redis_using_method(
+            in_key=in_key,
+            reading_method=self.conn.xrevrange,
+            name=in_key,
+            count=1,
+            min=self._add_offset_to_stream_id(self.last_msg_id, 1)
+        )
+
     def receive(self, in_key):
-        # TODO - refactor to use xread instead of xrevrange
-        redis_msg = self.conn.xrevrange(in_key, count=1)
+        # Need to set value in last_msg_id so
+        # _read_from_redis_using_method will not cause an infinite loop
+        self.last_msg_id = ""
+        return self._read_from_redis_using_method(
```

Comment:
The comment above doesn't sound good. It seems like last_msg_id should be a parameter given to `_read_from_redis_using_method`, in order for it to not be effected from the state of the class. What do you think?

Reply:
I agree with what you say but, `last_msg_id` can't be a parameter because almost all of the functions that read from the the redis, using this information and then will need to update `last_msg_id` by themselves. What i am saying is that changing it to a parameter will cause repetitive lines in many functions.